### PR TITLE
Add test for process_submission_message

### PR DESCRIPTION
**Pull request by matsujika**
- [The corresponding GCI task](https://codein.withgoogle.com/dashboard/task-instances/5118940880044032/)
- Changes
  - Add a unit test to confirm the function runs successfully when proper data exists.
  - Add a unit test to confirm the function returns `None` when submission does not exist.
  - Add a unit test to confirm the function raises exceptions when challenge phase does not exist.

### CODE REVIEW

File: tests/unit/remoteworker/test_remote_worker.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
 import mock
+import os
+import shutil
 
-from unittest import TestCase
+from django.core.files.uploadedfile import SimpleUploadedFile
+
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: remove newline

File: tests/unit/remoteworker/test_remote_worker.py [hunk 2]
```diff
@@ -1,6 +1,12 @@
...
+from rest_framework.test import APITestCase
+
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: remove newline

File: tests/unit/remoteworker/test_remote_worker.py [hunk 3]
```diff
@@ -141,3 +148,158 @@ def test_return_url_per_environment(self):
...
+    @mock.patch("scripts.workers.remote_submission_worker.SUBMISSION_DATA_DIR", "mocked/dir/submission_{submission_id}")
+    @mock.patch("scripts.workers.remote_submission_worker.os.path.basename", return_value="user_annotation_file.txt")
+    @mock.patch("scripts.workers.remote_submission_worker.get_submission_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_by_queue_name")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_phase_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.create_dir_as_python_package")
+    @mock.patch("scripts.workers.remote_submission_worker.download_and_extract_file")
+    @mock.patch("scripts.workers.remote_submission_worker.run_submission")
+    def test_process_submission_message_successfully(self, mock_run_submission,
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: can we rename this method to `test_process_submission_message` or `test_process_submission_message_success`

**Comment by takitsuse** [comment]
Changes are made as requested. @Ram81 please take a look.

**Comment by pushkalkatara** [comment]
@takitsuse Please fix the merge conflicts.

**Comment by takitsuse** [comment]
@pushkalkatara I resolved it. Please have a look

File: tests/unit/remoteworker/test_remote_worker.py [hunk 4]
```diff
@@ -5,7 +5,9 @@
 
 from os.path import join
 
-from unittest import TestCase
+from django.core.files.uploadedfile import SimpleUploadedFile
+from rest_framework.test import APITestCase
+from participants.models import Participant
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: @takitsuse can you reorder the imports lexicographically

File: tests/unit/remoteworker/test_remote_worker.py [hunk 5]
```diff
@@ -149,6 +152,161 @@ def test_return_url_per_environment(self):
...
+            self.challenge_pk,
+            self.challenge_phase,
+            self.submission,
+            user_annotation_file_path,
+            None
+        )
+
+    @mock.patch("scripts.workers.remote_submission_worker.extract_submission_data")
+    def test_process_submission_message_when_submission_does_not_exist(self, mock_esd):
+        self.submission_pk += 1
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: @takitsuse rather than incrementing the actual variable can you use `self.submission_pk + 1` wherever it is needed. As this is a test for invalid submission we shouldn't update the actual  `self.submission_pk`

File: tests/unit/remoteworker/test_remote_worker.py [hunk 6]
```diff
@@ -149,6 +152,161 @@ def test_return_url_per_environment(self):
...
+
+    @mock.patch("scripts.workers.remote_submission_worker.extract_submission_data")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_by_queue_name")
+    @mock.patch("scripts.workers.remote_submission_worker.get_challenge_phase_by_pk")
+    @mock.patch("scripts.workers.remote_submission_worker.logger.exception")
+    def test_process_submission_message_when_challenge_phase_does_not_exist(self, mock_logger,
+                                                                            mock_get_challenge_phase_by_pk,
+                                                                            mock_get_challenge_by_queue_name,
+                                                                            mock_esd):
+        self.challenge_phase_pk += 1
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: Same here

**Comment by pushkalkatara** [comment]
Looks good to me!

File: tests/unit/remoteworker/test_remote_worker.py [hunk 7]
```diff
@@ -5,7 +5,9 @@
 
 from os.path import join
 
-from unittest import TestCase
+from rest_framework.test import APITestCase
+from participants.models import Participant
+from django.core.files.uploadedfile import SimpleUploadedFile
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: @takitsuse `from django.core.files.uploadedfile import SimpleUploadedFile` should be at 8th line

Comment (takitsuse) [reply tests/unit/remoteworker/test_remote_worker.py]: In this order?:
```python
from django...
from rest_...
from participants...
```

**Comment by takitsuse** [comment]
I reordered imports using https://github.com/Cloud-CV/EvalAI/blob/master/tests/unit/worker/test_submission_worker.py#L11-L31 as a reference. @Ram81 Please take a look.

File: tests/unit/remoteworker/test_remote_worker.py [hunk 8]
```diff
@@ -5,8 +5,11 @@
 
 from os.path import join
 
-from unittest import TestCase
+from django.core.files.uploadedfile import SimpleUploadedFile
 
+from rest_framework.test import APITestCase
+
+from participants.models import Participant
```
Comment (Ram81) [inline tests/unit/remoteworker/test_remote_worker.py]: @takitsuse remove the new line change the imports on line 10 and 11 to 

```
from participants.models import Participant
from rest_framework.test import APITestCase
```

Imports should be in lexicographical order
